### PR TITLE
Serato markers improvements

### PR DESCRIPTION
**Pull request by Holzhaus**
Some code cleanup, improved comments and member visibility improvements.

### CODE REVIEW

**Comment by Holzhaus** [comment]
Oof, this finally adds the missing test for hotcue/loop parsing in FLAC data. Parsing did already work before, but due to Serato's weird base64 writer we were unable to test it because the regular roundtrip tests didn't work due to differences in the last byte. This finally fixes the issue by storing the last byte in the SeratoMarkers2 instance.

File: src/track/serato/markers.h [hunk 1]
```diff
@@ -120,36 +120,30 @@ inline QDebug operator<<(QDebug dbg, const SeratoMarkersEntry& arg) {
...
+/// This class includes functions for formatting and parsing SeratoMarkers_
+/// metadata according to the specification:
+/// https://github.com/Holzhaus/serato-tags/blob/master/docs/serato_markers_.md
 class SeratoMarkers final {
   public:
-    SeratoMarkers() = default;
-    explicit SeratoMarkers(QList<SeratoMarkersEntryPointer> entries)
-            : m_entries(std::move(entries)) {
-    }
+    SeratoMarkers(){};
```
Comment (uklotzde) [inline src/track/serato/markers.h:131]: `= default` is longer but more specific/explicit then a a custom, empty implementation.

File: src/track/serato/markers2.h [hunk 2]
```diff
@@ -434,7 +432,38 @@ class SeratoMarkers2 final {
...
+    static bool parseID3(
+            SeratoMarkers2* seratoMarkers2,
+            const QByteArray& outerData);
+    /// The Base64 encoded format is used for MP4 tags and (except for a single
+    /// byte) for FLAC tags.
+    static bool parseBase64Encoded(
+            SeratoMarkers2* seratoMarkers2,
+            const QByteArray& base64EncodedData);
+    /// The FLAC format is basically identical to MP4, but Serato seems to
+    /// randomly modify the last byte of the base64 encoding. This this will
```
Comment (uklotzde) [inline src/track/serato/markers2.h:447]: Typo: This this

File: src/track/serato/markers.cpp [hunk 3]
```diff
@@ -164,7 +166,9 @@ SeratoMarkersEntryPointer SeratoMarkersEntry::parseID3(const QByteArray& data) {
         // End position not set
         if (endPositionSerato32 != 0x7F7F7F7F) {
```
Comment (uklotzde) [inline src/track/serato/markers.cpp:167]: Encode this literal as a constant and format as hex in the log message?

File: src/test/seratomarkers2test.cpp [hunk 4]
```diff
@@ -12,7 +12,7 @@ namespace {
 
 class SeratoMarkers2Test : public testing::Test {
   protected:
-    void parseBpmlockEntry(const QByteArray inputValue, bool valid, bool locked) {
+    void parseBpmlockEntry(const QByteArray& inputValue, bool valid, bool locked) {
```
Comment (uklotzde) [inline src/test/seratomarkers2test.cpp:15]: Renaming? -> parseBpmLockEntry

**Review by uklotzde — COMMENTED** [review]
Some minor comments. Impressive re-engineering work!

These weird inconsistencies shed some light on Serato's engineering competence and code quality.

**Comment by Holzhaus** [comment]
Thanks for reviewing!

> These weird inconsistencies shed some light on Serato's engineering competence and code quality.

Yup. I'm pretty sure some of these random bytes are caused by messed up length checks that lead to writing uninitialized memory. Otherwise I can't explain these random values at the end...

**Review by uklotzde — APPROVED** [review]
LGTM

**Pull request merged**